sgx/common/src/ecdh_util.c

```diff
@@ -0,0 +1,343 @@
+/**
+ * @file ecdh_util.c
+ *
+ * @brief Provides implementation for functionality directly supporting
+ *        elliptic curve Diffe-Hellman key agreement within SGX applications
+ *        employing kmyth.
+ */
+
+#include "ecdh_util.h"
+
+/*****************************************************************************
+ * create_ecdh_ephemeral_key_pair()
+ ****************************************************************************/
+int create_ecdh_ephemeral_key_pair(int ec_nid,
+                                   EC_KEY ** ephemeral_ec_key_pair_out)
+{
+  // create new EC_KEY object for the specified built-in curve
+  //   The EC_KEY object passed to 'generate_key' below must be associated
+  //   with the desired EC_GROUP.
+  *ephemeral_ec_key_pair_out = EC_KEY_new_by_curve_name(ec_nid);
+  if (*ephemeral_ec_key_pair_out == NULL)
+  {
+    kmyth_sgx_log(3, "failed to create new elliptic curve key object by NID");
```

Comment:
Is there a reason not to include syslog.h for the log severity (e.g., LOG_ERR)?

Reply:
syslog.h did not seem to be available inside the enclave, so I moved on when I was working on this. I will look into this further as I agree the magic numbers for things like log message severity levels is suboptimal.

Reply:
In the most recent commit (9045fff), I think I addressed the use of the logging priority macros defined in syslog.h. For now, if they are not already defined, I define these macros in sgx/common/include/kmyth_enclave_common.h. Longer term, there is probably a more elegant way to include them directly from syslog.h inside the enclave.